src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -335,10 +335,10 @@ private TypeSymbolWithAnnotations GetTypeSymbol()
                 // If we got a valid result that was not void then use the inferred type
                 // else create an error type.
                 if ((object)inferredType != null &&
-                    inferredType.SpecialType != SpecialType.System_Void)
+                    inferredType.TypeSymbol.SpecialType != SpecialType.System_Void)
```

Comment:
This change looks unnecessary, TypeSymbolWithAnnotations has SpecialType property?